src/utils/XFDR.cpp

```diff
@@ -0,0 +1,1095 @@
+      {
+        types.push_back(TOPPXFDR::xlclass_intradecoys);
+      }
+
+      // Decoys
+      if (pep_is_decoy)
+      {
+        types.push_back(TOPPXFDR::xlclass_decoys);
+      }
+
+      // intralinks
+      if (pep_id.metaValueExists("OpenXQuest:is_intraprotein") && ! pep_is_decoy)
+      {
+        types.push_back(TOPPXFDR::xlclass_intralinks);
+      }
+
+      // interdecoys
+      if (pep_id.metaValueExists("OpenXQuest:is_interprotein") && pep_is_decoy)
+      {
+        types.push_back(TOPPXFDR::xlclass_interdecoys);
+      }
+
+      // interlinks
+      if (pep_id.metaValueExists("OpenXQuest:is_interprotein") && ! pep_is_decoy)
+      {
+        types.push_back(TOPPXFDR::xlclass_interlinks);
+      }
+      String xl_type = getXLMetaValue<String>("xl_type", pep_id, false);
+
+      // monolinks
+      if ( ! pep_is_decoy && (xl_type == "mono-link"
+                              ||  xl_type == "loop-link"))
+      {
+        types.push_back(TOPPXFDR::xlclass_monolinks);
+      }
+
+      // monodecoys
+      if ( pep_is_decoy && (xl_type == "mono-link"
+                            ||  xl_type == "loop-link"))
+      {
+        types.push_back(TOPPXFDR::xlclass_monodecoys);
+      }
+      const vector< PeptideHit > & pep_hits = pep_id.getHits();
+      if (pep_hits.size() == 2)
+      {
+        PeptideHit alpha = pep_hits[0];
+        PeptideHit beta = pep_hits[1];
+        
+        bool alpha_is_decoy = alpha.getMetaValue("target_decoy").toString() == "decoy";
+        bool beta_is_decoy = beta.getMetaValue("target_decoy").toString() == "decoy";
+
+        // fulldecoysintralinks
+        if (   pep_id.metaValueExists("OpenXQuest:is_intraprotein")
+               && alpha_is_decoy
+               && beta_is_decoy)
+        {
+          types.push_back(TOPPXFDR::xlclass_fulldecoysintralinks);
+        }
+
+        // fulldecoysinterlinks
+        if (   pep_id.metaValueExists("OpenXQuest:is_interprotein")
+               && alpha_is_decoy
+               && beta_is_decoy)
+        {
+          types.push_back(TOPPXFDR::xlclass_fulldecoysinterlinks);
+
+        }
+
+        // hybriddecoysintralinks
+        if (       pep_id.metaValueExists("OpenXQuest:is_intraprotein")
+                   && (( ! alpha_is_decoy
+                        &&     beta_is_decoy)
+                        ||     (alpha_is_decoy
+                        &&   ! beta_is_decoy)))
+        {
+          types.push_back(TOPPXFDR::xlclass_hybriddecoysintralinks);
+        }
+
+        // hybriddecoysinterlinks
+        if (       pep_id.metaValueExists("OpenXQuest:is_interprotein")
+                   && (( ! alpha_is_decoy
+                        &&     beta_is_decoy)
+                        ||     (alpha_is_decoy
+                        &&   ! beta_is_decoy)))
+        {
+          types.push_back(TOPPXFDR::xlclass_hybriddecoysinterlinks);
+        }
+      }
+    }
+
+
+    /** False positve counting as performed by xProphet software package.
```

Comment:
please remove big bulks of commented out code (e.g., you can add a commit just for removing the comments. That way you easily find it again if you need to)